pkg/agent/manager/manager_test.go

```diff
@@ -361,9 +359,9 @@ func TestSynchronization(t *testing.T) {
 	util.RunWithTimeout(t, 2*m.syncFreq, func() {
 		// There should be 3 updates after sync, because we are subcribed to selectors that
 		// matches with 3 entries that were renewed on the cache.
-		<-wu
-		<-wu
-		u := <-wu
+		<-sub.Updates()
+		<-sub.Updates()
+		u := <-sub.Updates()
```

Comment:
we still get multiple updates here because... we apply 3 changes to cache, and notify on each one of those changes?

Reply:
Yeah, notification is still happening any time we set a new cache entry (or replace an expired one), so 3 expired entries, 3 new notifications. I think this should be tracked with another issue. I remember you mentioned this already [here](https://github.com/spiffe/spire/pull/441/files/bc2ec80d58efbe39cad3c5a2f9b9b6b1a1246160#diff-a2625dd10ec70286304fbd610f3081e2R377).